src/test/coins_tests.cpp

```diff
@@ -764,54 +771,40 @@ static void CheckAddCoinBase(CAmount base_value, CAmount modify_value, const Coi
+        BOOST_CHECK_EQUAL(FAIL_NO_ENTRY, expected);
     }
 }
-static void CheckAddCoinBase(CAmount base_value, CAmount cache_value, CAmount modify_value, CAmount expected_value, char cache_flags, char expected_flags, bool coinbase)
-{
-    CheckAddCoinBase(base_value, modify_value, CoinEntry{cache_value, cache_flags}, CoinEntry{expected_value, expected_flags}, coinbase);
-}
-
-// Simple wrapper for CheckAddCoinBase function above that loops through
-// different possible base_values, making sure each one gives the same results.
-// This wrapper lets the coins_add test below be shorter and less repetitive,
-// while still verifying that the CoinsViewCache::AddCoin implementation
-// ignores base values.
-template <typename... Args>
-static void CheckAddCoin(Args&&... args)
-{
-    for (const CAmount base_value : {ABSENT, SPENT, VALUE1})
-        CheckAddCoinBase(base_value, std::forward<Args>(args)...);
-}
 
 BOOST_AUTO_TEST_CASE(ccoins_add)
 {
     /* Check AddCoin behavior, requesting a new coin from a cache view,
      * writing a modification to the coin, and then checking the resulting
      * entry in the cache after the modification. Verify behavior with the
-     * AddCoin possible_overwrite argument set to false, and to true.
-     *
-     *           Cache   Write   Result  Cache        Result       possible_overwrite
-     *           Value   Value   Value   Flags        Flags
+     * AddCoin coinbase argument set to false, and to true.
+     *               Base        Write   Cache               Expected            Coinbase
      */
-    CheckAddCoin(ABSENT, VALUE3, VALUE3, NO_ENTRY   , DIRTY|FRESH, false);
-    CheckAddCoin(ABSENT, VALUE3, VALUE3, NO_ENTRY   , DIRTY      , true );
-    CheckAddCoin(SPENT , VALUE3, VALUE3, 0          , DIRTY|FRESH, false);
-    CheckAddCoin(SPENT , VALUE3, VALUE3, 0          , DIRTY      , true );
-    CheckAddCoin(SPENT , VALUE3, VALUE3, FRESH      , DIRTY|FRESH, false);
-    CheckAddCoin(SPENT , VALUE3, VALUE3, FRESH      , DIRTY|FRESH, true );
-    CheckAddCoin(SPENT , VALUE3, VALUE3, DIRTY      , DIRTY      , false);
-    CheckAddCoin(SPENT , VALUE3, VALUE3, DIRTY      , DIRTY      , true );
-    CheckAddCoin(SPENT , VALUE3, VALUE3, DIRTY|FRESH, DIRTY|FRESH, false);
-    CheckAddCoin(SPENT , VALUE3, VALUE3, DIRTY|FRESH, DIRTY|FRESH, true );
-    CheckAddCoin(VALUE2, VALUE3, FAIL  , 0          , NO_ENTRY   , false);
-    CheckAddCoin(VALUE2, VALUE3, VALUE3, 0          , DIRTY      , true );
-    CheckAddCoin(VALUE2, VALUE3, FAIL  , FRESH      , NO_ENTRY   , false);
-    CheckAddCoin(VALUE2, VALUE3, VALUE3, FRESH      , DIRTY|FRESH, true );
-    CheckAddCoin(VALUE2, VALUE3, FAIL  , DIRTY      , NO_ENTRY   , false);
-    CheckAddCoin(VALUE2, VALUE3, VALUE3, DIRTY      , DIRTY      , true );
-    CheckAddCoin(VALUE2, VALUE3, FAIL  , DIRTY|FRESH, NO_ENTRY   , false);
-    CheckAddCoin(VALUE2, VALUE3, VALUE3, DIRTY|FRESH, DIRTY|FRESH, true );
+    for (auto base_value : {ABSENT, SPENT, VALUE1}) {
```

Comment:
In commit "test: Group values and states in tests into CoinEntry wrappers" (553db5dec1334ed5e8e07e34306f16de9afc22bb)

Note: because this for loop is moving from `CheckAddCoin` function to the top-level test code, this commit changes order of the tests. But tests that are performed should still be the same. It would be good to note in commit message that this commit does affect test order of this test but doesn't change its coverage.